rrtmgp/kernels/mo_reorder_kernels.F90

```diff
@@ -24,18 +24,34 @@ subroutine reorder_123x312_kernel(d1, d2, d3, array_in, array_out) &
     real(wp), dimension(d1, d2, d3), intent( in) :: array_in
     real(wp), dimension(d3, d1, d2), intent(out) :: array_out
 
-    integer :: i1, i2, i3
+    integer :: i1, i2, i3, i10, i30, i1diff, i3diff
+    integer, parameter :: tile = 32
 
-    !$acc parallel loop collapse(3) &
-    !$acc&     copyout(array_out(:d3,:d1,:d2)) &
-    !$acc&     copyin(array_in(:d1,:d2,:d3))
+
+    !$acc parallel vector_length(tile*tile) &
+    !$acc&     copyout(array_out) &
+    !$acc&     copyin(array_in)
+    !$acc loop gang collapse(3)
     do i2 = 1, d2
```

Comment:
The code is substantially less clear than a simple transposition. That's ok, but again please add a few lines of comments explaining that you're breaking the array up into pieces to do the transposition?

Reply:
The two inner loops ensure that we're reading and writing the data in blocks of `tile*tile`. So the outer loops encode the blocks, while the inner loops go over all the elements in each of the blocks. That helps keeping the elements in such a block completely in the fastest cache.